app/routes/__index/domains/index.tsx

```diff
@@ -15,20 +20,51 @@ export const loader = async ({ request }: LoaderArgs) => {
   return typedjson(await getRecordsByUsername(username));
 };
 
+export const action = async ({ request }: ActionArgs) => {
+  const { id } = Object.fromEntries(await request.formData());
+  const formalizedID = Number(id);
+
+  if (request.method === 'PUT') {
+    const record = await getRecordById(formalizedID);
+    if (record) {
+      const newExpiresAt = new Date(record.expiresAt);
+      newExpiresAt.setMonth(newExpiresAt.getMonth() + 6);
+
+      await updateRecordById(
+        Number(id),
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        newExpiresAt
+      );
+    }
+  } else if (request.method === 'DELETE') {
+    await deleteRecordById(formalizedID);
```

Comment:
If I understand this correctly, this is now what we should do.

If you take the record and delete it from the FE, that will not actually delete the record from route53, it will only delete our knowledge of it.

Instead, we should kick off a worker, that removes the record from route53, then delete it from the db as well

Reply:
I feel like we don't have an api for making a proper deletion yet, I guess it makes sense to wait till some related PRs gonna be merged as like #274 as @humphd mentioned

Reply:
To be honest, I'd just add a TODO into the code instead of this DB tie and ship this PR.
If a PR is not merged, it's constant work to resolve conflicts and keep it up to date and ready to merge while others are working on the same piece of code

Reply:
@dadolhay makes sense, I did what you say, just left todos comments and opened issues to resolve them when the appropriate code for all these things gonna be shipped #280 #281